### ISSUE TITLE
numa and knl locale models have functionality commented out, citing a retired future

### ISSUE THREAD

**Issue by cassella** [chapel-lang/chapel#8508]
### Summary of Problem

In the numa and knl locale model code,

```
modules/internal/localeModels/numa/LocaleModel.chpl
modules/internal/localeModels/knl/LocaleModel.chpl
```

there's a ``NumaDomain.getChildren()`` iterator commented, out citing a future that's been retired:

```chapel
    // This is commented out b/c it leads to an internal error during
    // the resolveIntents pass.  See
    // test/functions/iterators/sungeun/iterInClass.future
    //
    // iter getChildren() : locale {
    //  halt("No children to iterate over.");
    //  yield nil;
    // }
```

That future was retired in dc11834584f729ff0f2f91ad755d561f4016d2db

**Comment by lydia-duncan** [comment]
@dmk42 or @gbtitus - would one of you look at this as a clean up item?  I could take a look myself but I suspect there is context I don't have

**Comment by dmk42** [comment]
This was closed by #8898 .
